R/factory_root.R

```diff
@@ -0,0 +1,53 @@
+#' Find the path to the root folder within a factory
+#'
+#' This function is used to find the fully expanded root path within a factory.
+#'
+#' @param directory the full path to the directory you are interested in.  If
+#'   NULL (default) the function assumes you are already in a factory and will
+#'   return the address of the containing factory root.
+#'
+#' @noRd
+#' @keywords internal
+factory_root <- function(directory = NULL) {
+  
+  if(!is.null(directory)) {
```

Comment:
I'm probably thick here, but would it be simpler to start off with something like:
```

if (is.null(directory)) {
  directory <- getwd()
}
```
or even define `directory = getwd()` in the arguments?

Reply:
Yes - no idea why I did this.  Using `"."` over getwd() as it seems like one less function call.  closed with https://github.com/reconhub/reportfactory/pull/68/commits/e60198f05faaf899a33aab96f3d9634ab3413252